crates/aptos-indexer-api/src/main.rs

```diff
@@ -0,0 +1,41 @@
+mod graphql_root_queries;
+
+use aptos_indexer::database::new_db_pool;
+use async_graphql::{
+    http::{playground_source, GraphQLPlaygroundConfig},
+    EmptyMutation, EmptySubscription, Schema,
+};
+use async_graphql_poem::GraphQL;
+use clap::Parser;
+use graphql_root_queries::{ContextData, QueryRoot};
+use poem::{get, handler, listener::TcpListener, web::Html, IntoResponse, Route, Server};
```

Comment:
does poem give us any benefits here over the other options?

Reply:
At this level it probably doesn't make much of a difference, it's just a thin layer on top of hyper. I need to do my own benchmarking between warp and poem still though, so TBA.

Reply:
Idk if we'll need to make a REST API but it could come in handy. Honestly it'd be pretty trivial to remove it so I'm inclined to keep ti for now.